### PR TITLE
docs(FloatButtonGroup): Add onClick prop

### PR DESCRIPTION
**Pull request by Riyad-Arafat**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      fix: missing onClick in FloatButtonGroup    |
| 🇨🇳 Chinese |       修复: FloatButtonGroup 遗漏  onClick   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/float-button/index.en-US.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ FloatButton. Available since `5.0.0`.
 | trigger | Which action can trigger menu open/close | `click` \| `hover` | - |  |
 | open | Whether the menu is visible or not | boolean | - |  |
 | onOpenChange | Callback executed when active menu is changed | (open: boolean) => void | - |  |
+| onClick | Set the handler to handle `click` event | (event) => void | - |  |
```
Comment (li-jia-nan) [inline components/float-button/index.en-US.md:55]:
```suggestion
| onClick | Set the handler to handle `click` event (only work in `Menu mode`) | (event) => void | - |  |
```

File: components/float-button/index.zh-CN.md [hunk 2]
```diff
@@ -53,6 +53,7 @@ demo:
 | trigger      | 触发方式（有触发方式为菜单模式） | `click` \| `hover`      | -        |      |
 | open         | 受控展开                         | boolean                 | -        |      |
 | onOpenChange | 展开收起时的回调                 | (open: boolean) => void | -        |      |
+| onClick      | 点击按钮时的回调                 | (event) => void         | -        |      |
```
Comment (li-jia-nan) [inline components/float-button/index.zh-CN.md:56]:
```suggestion
| onClick      | 点击按钮时的回调（仅在菜单模式中有效）                 | (event) => void         | -        |      |
```

**Comment by MadCcc** [comment]
Any senario in real world needs this prop? Is `onOpenChange` enough?

**Comment by riyadelberkawy** [comment]
@MadCcc 
The reason I added this prop is to enable the ability to update the `open` prop when clicking on the button. In some cases, this behavior may be required, and the `onClick` prop provides a convenient way to handle it.

**Comment by MadCcc** [comment]
> @MadCcc The reason I added this prop is to enable the ability to update the `open` prop when clicking on the button. In some cases, this behavior may be required, and the `onClick` prop provides a convenient way to handle it.

In this case `onOpenChange` and `trigger="click"` is enough.

**Comment by riyadelberkawy** [comment]
> > @MadCcc The reason I added this prop is to enable the ability to update the `open` prop when clicking on the button. In some cases, this behavior may be required, and the `onClick` prop provides a convenient way to handle it.
> 
> In this case `onOpenChange` and `trigger="click"` is enough.

You're correct, but `FloatButtonGroupProps` extends `FloatButtonProps`. I became confused when I attempted to use it and it failed to work.

**Comment by riyadelberkawy** [comment]
@MadCcc 
In my case, I want the menu to remain open even when I click away from the button. I want the open state to be changed only when the float button is clicked.

**Comment by li-jia-nan** [comment]
We worry it is easy to cause ambiguity, some people will think that the click of the FloatButtonGroup is the entire area, and some people will think it is the trigger button.

**Comment by riyadelberkawy** [comment]
> We worry it is easy to cause ambiguity, some people will think that the click of the FloatButtonGroup is the entire area, and some people will think it is the trigger button.

Thank you for bringing up the concern about ambiguity. I agree that choosing a clear and meaningful name for the click area would be the best solution to avoid any confusion.
Do you have any suggestions for a more descriptive name that we could use instead?

**Review by aiden112286 — APPROVED** [review]

**Review by AbdullahWins — APPROVED** [review]
good job!

**Comment by afc163** [comment]
Conflicting

**Comment by riyadelberkawy** [comment]
@afc163 
resolved

**Comment by afc163** [comment]
lint 挂了，看看

**Comment by Riyad-Arafat** [comment]
> lint 挂了，看看

Checking

File: components/float-button/__tests__/group.test.tsx [hunk 3]
```diff
@@ -86,6 +86,37 @@ describe('FloatButtonGroup', () => {
...
+        <FloatButton onClick={onClick2} />
+        <FloatButton onClick={onClick2} />
+        <FloatButton onClick={onClick2} />
+      </FloatButton.Group>,
+    );
+    fireEvent.click(container.querySelector('.ant-float-btn')!);
+    fireEvent.click(container.querySelector('.ant-float-btn-group')!);
+    expect(onClick).toHaveBeenCalled();
+    expect(onClick2).not.toHaveBeenCalled();
+   });
```
Comment (afc163) [inline components/float-button/__tests__/group.test.tsx:119]:
```suggestion
   });

```

**Comment by Riyad-Arafat** [comment]
> Is this just a documentation and test case supplement? Has `onClick` actually been supported since version 5.3.0?
> 
> If so, this PR can be sent directly to the master branch and prefixed with `docs:`.

Yes, this PR was created long before this release.

**Review by afc163 — APPROVED** [review]

**Pull request merged**